frontend/src/hooks/useGetBoardInfo.ts

```diff
@@ -55,27 +57,52 @@ export const useGetBoardInfoEffect = (): useGetBoardInfoInterface => {
   const [isLoading, setIsLoading] = useState(false);
   const [errorMessage, setErrorMessage] = useState({});
 
-  const getBoardInfo = (data: BoardInfoRequestDTO) => {
+  const getBoardInfo = async (data: BoardInfoConfigDTO) => {
     setIsLoading(true);
     setErrorMessage({});
-    return boardInfoClient
-      .getBoardInfo(data)
-      .then((res) => {
-        if (!res.data) {
-          setErrorMessage({
-            title: BOARD_CONFIG_INFO_TITLE.NO_CONTENT,
-            message: BOARD_CONFIG_INFO_ERROR.NOT_CONTENT,
-            code: HttpStatusCode.NoContent,
+
+    if (data.dateRanges) {
+      const dateRangeCopy = Array.from(data.dateRanges);
+      dateRangeCopy.sort((a, b) => dayjs(a.startDate).valueOf() - dayjs(b.startDate).valueOf());
+      const allBoardData = dateRangeCopy.map((info) => {
+        const request = {
+          token: data.token,
+          type: data.type,
+          site: data.site,
+          email: data.email,
+          boardId: data.boardId,
+          projectKey: data.projectKey,
+        };
+        const boardInfoRequest = {
+          ...request,
+          startTime: dayjs(info.startDate).valueOf().toString(),
+          endTime: dayjs(info.endDate).valueOf().toString(),
+        };
+
+        return boardInfoClient
+          .getBoardInfo(boardInfoRequest)
+          .then((res) => {
+            if (!res.data) {
+              setErrorMessage({
+                title: BOARD_CONFIG_INFO_TITLE.NO_CONTENT,
+                message: BOARD_CONFIG_INFO_ERROR.NOT_CONTENT,
+                code: HttpStatusCode.NoContent,
+              });
+            }
+            return res;
+          })
+          .catch((err) => {
+            const { code } = err;
+            setErrorMessage(codeMapping(code));
+            return err;
           });
-        }
+      });
+
+      return await Promise.all(allBoardData).then((res) => {
```

Comment:
1. How to handle the `catch/failed` part with `Promise.all`, and the `setIsLoading(false);` when fialed
2. you use `await`, why did you use `.then()`